sparsh2/CS-555-Senior-Sage
# SCRUM-42: App Development Setup

This PR sets up react-native based app development project structure. JIRA for this PR: [SCRUM-42](https://team17seniorsage.atlassian.net/browse/SCRUM-42?atlOrigin=eyJpIjoiODYyZTBjMWIyOTlkNDQzOGFkZmMyZTJiNTdmMDIzYWIiLCJwIjoiaiJ9)

How to test?
- Install Java, Android Studio, Android SDK, nodejs
  - Checkout this guide from official react-native website for more detailed guide to set up your environment: https://reactnative.dev/docs/set-up-your-environment
  - Additional resource: [Microsoft's React Native for Android tutorial](https://learn.microsoft.com/en-us/windows/dev-environment/javascript/react-native-for-android)
- Checkout this PR on local
  - `git fetch origin pull/2/head:pr/2`
- Navigate to `app/` dir
- Run `npm install` to install all the dependencies for the project
- Connect your android device to laptop (or use emulators in android studio)
  - Ensure you have enabled debugging mode on your android phone
  - Run `adb devices` and ensure you are seeing your device connected before installing the app
- Run `npx react-native run-android` to build and install the app on your device
